datastore/src/main/java/io/spine/server/storage/datastore/DsProjectionStorageDelegate.java

```diff
@@ -0,0 +1,142 @@
+import io.spine.server.entity.storage.ColumnTypeRegistry;
+import io.spine.server.storage.datastore.type.DatastoreColumnType;
+import io.spine.type.TypeUrl;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.server.storage.datastore.DsFilters.activeEntity;
+
+/**
+ * A {@link io.spine.server.storage.RecordStorage RecordStorage} to which
+ * {@link DsProjectionStorage} delegates its operations.
+ *
+ * @author Dmytro Dashenkov
+ */
+public class DsProjectionStorageDelegate<I> extends DsRecordStorage<I> {
+
+    protected DsProjectionStorageDelegate(
+            Descriptor descriptor,
+            DatastoreWrapper datastore,
+            boolean multitenant,
+            ColumnTypeRegistry<? extends DatastoreColumnType<?, ?>> columnTypeRegistry,
+            Class<I> idClass) {
+        super(descriptor, datastore, multitenant, columnTypeRegistry, idClass);
+    }
+
+    private DsProjectionStorageDelegate(Builder<I> builder) {
+        this(builder.descriptor,
+             builder.datastore,
+             builder.multitenant,
+             builder.columnTypeRegistry,
+             builder.idClass);
+    }
+
+    @Override
+    protected StructuredQuery<Entity> buildAllQuery(TypeUrl typeUrl) {
+        final String entityKind = kindFrom(typeUrl).getValue();
+        final StructuredQuery<Entity> query = Query.newEntityQueryBuilder()
+                                                   .setKind(entityKind)
+                                                   .setFilter(activeEntity())
+                                                   .build();
+        return query;
+    }
+
+    public static <I> Builder<I> builder() {
+        return new Builder<>();
+    }
+
+    /**
+     * A builder for the {@code DsProjectionStorageDelegate}.
+     */
+    public static class Builder<I> {
+
+        private Descriptor descriptor;
+        private DatastoreWrapper datastore;
+        private boolean multitenant;
+        private ColumnTypeRegistry<? extends DatastoreColumnType<?, ?>> columnTypeRegistry;
+        private Class<I> idClass;
+
+        private Builder() {
+            // Prevent direct initialization.
+        }
+
+        public Builder<I> setStateType(TypeUrl stateTypeUrl) {
+            checkNotNull(stateTypeUrl);
+            final Descriptor descriptor = (Descriptor) stateTypeUrl.getDescriptor();
+            this.descriptor = checkNotNull(descriptor);
+            return this;
+        }
+
+        /**
+         * @param datastore the {@link DatastoreWrapper} to use in this storage
+         */
+        public Builder<I> setDatastore(DatastoreWrapper datastore) {
+            this.datastore = checkNotNull(datastore);
+            return this;
+        }
+
+        /**
+         * @param multitenant {@code true} if the storage should be
+         *                    {@link io.spine.server.storage.Storage#isMultitenant multitenant}
+         *                    or not
```

Comment:
`,` at the end of the previous line `{@code false} otherwise` should go instead of this line. Right now it makes not much sense.